iradix.go

```diff
@@ -384,6 +460,20 @@ func (t *Txn) Delete(k []byte) (interface{}, bool) {
 	return nil, false
 }
 
+// DeletePrefix is used to delete an entire subtree that matches the prefix
+// This will delete all nodes under that prefix
+func (t *Txn) DeletePrefix(prefix []byte) bool {
+	newRoot, numDeletions := t.deletePrefix(nil, t.root, prefix)
+	if newRoot != nil {
+		t.root = newRoot
+	}
+	if numDeletions > 0 {
```

Comment:
This doesn't seem like it can be true unlees `newRoot != nil`, can this just be:

```
if newRoot != nil {
    t.root = newRoot
   t.size = t.size - numDeletions
}
```